### PR TITLE
Added About menu

### PR DESCRIPTION
**Pull request by alwayslivid**
Relevant issue: #4

### CODE REVIEW

File: source/MainWindow.cpp [hunk 1]
```diff
@@ -95,6 +96,7 @@ MainWindow::MainWindow(BRect frame, BEntry* assocEntry, SettingsFile* settings)
 	fSettingsItem = new BMenuItem("Settings", new BMessage(MSG_SETTINGS));
 
 	fHelpMenu = new BMenu("Help", B_ITEMS_IN_COLUMN);
+	fAboutItem = new BMenuItem("About", new BMessage(B_ABOUT_REQUESTED));
```
Comment (humdingerb) [inline source/MainWindow.cpp]: There doesn't seem a standard for this, but I think it's nicer to have "About ResourceEdit" for the label.

File: source/MainWindow.cpp [hunk 2]
```diff
@@ -234,6 +237,24 @@ MainWindow::~MainWindow()
 	delete fUndoContext;
 }
 
+void
+MainWindow::AboutRequested()
+{
+	BAboutWindow* about = new BAboutWindow("ResourceEdit","x-vnd.Haiku-ResourceEdit");
+	
+	const char* authors[] = {
+		"Tri-Edge AI",
```
Comment (humdingerb) [inline source/MainWindow.cpp]: Alpha-sort the authors.

File: source/MainWindow.cpp [hunk 3]
```diff
@@ -234,6 +237,24 @@ MainWindow::~MainWindow()
 	delete fUndoContext;
 }
 
+void
+MainWindow::AboutRequested()
+{
+	BAboutWindow* about = new BAboutWindow("ResourceEdit","x-vnd.Haiku-ResourceEdit");
```
Comment (humdingerb) [inline source/MainWindow.cpp]: Needs to be "application/x-vnd.Haiku-ResourceEdit" or the about window won't find its icon.

File: source/MainWindow.cpp [hunk 4]
```diff
@@ -234,6 +237,24 @@ MainWindow::~MainWindow()
...
+	
+	const char* authors[] = {
+		"Tri-Edge AI",
+		"waddlesplash",
+		"Panagiotis Vasilopoulos",
+		"Artur Jamro",
+		NULL
+	};
+
+	about->AddCopyright(2012-2013, "Tri-Edge AI");
```
Comment (humdingerb) [inline source/MainWindow.cpp]: "2012-2013" = -1 :)
Just add the start year, i.e. 2012.

File: source/ResourceEdit.cpp [hunk 5]
```diff
@@ -101,8 +99,15 @@ ResourceEdit::MessageReceived(BMessage* msg)
 			break;
 		}
 		case MSG_SETTINGS_CLOSED:
+		{
 			fSettingsWindow = NULL;
 			break;
+		}
+		case B_ABOUT_REQUESTED:
```
Comment (humdingerb) [inline source/ResourceEdit.cpp]: The B_ABOUT_REQUESTED case belongs into the MainWindow's MessageReceived().

File: source/ResourceEdit.h [hunk 6]
```diff
@@ -12,19 +12,19 @@
 class MainWindow;
 class SettingsFile;
 class SettingsWindow;
+class BAboutWindow;
```
Comment (humdingerb) [inline source/ResourceEdit.h]: Remove.

File: source/ResourceEdit.h [hunk 7]
```diff
@@ -12,19 +12,19 @@
...
 
-
 class ResourceEdit : public BApplication {
 public:
 					ResourceEdit();
 					~ResourceEdit();
 
 	void			MessageReceived(BMessage* msg);
-
+//	void			AboutRequested();
```
Comment (humdingerb) [inline source/ResourceEdit.h]: Remove.

File: source/MainWindow.cpp (hunk 4 above)
Comment (alwayslivid) [reply source/MainWindow.cpp]: Should've seen that one coming, my bad.

**Comment by humdingerb** [comment]
Thank you!

**Pull request merged**